lib/pliny/librato/metrics/backend.rb

```diff
@@ -54,23 +54,20 @@ def start_timer
           end
         end
 
-        def start_counter
-          @counter = Thread.new do
-            loop do
-              msg = metrics_queue.pop
-              msg == POISON_PILL ? break : enqueue_librato(msg)
+        def flush_librato
+          sync do
```

Comment:
@cyx now that https://github.com/librato/librato-metrics/pull/137 has landed maybe we can try backing off on some of these sync blocks?